cmake/ccache.cmake

```diff
@@ -0,0 +1,43 @@
+# Adapted from https://crascit.com/2016/04/09/using-ccache-with-cmake/
+cmake_minimum_required(VERSION 3.4)
+
+find_program(CCACHE_PROGRAM ccache)
+if(CCACHE_PROGRAM)
+  # Set up wrapper scripts
+  set(C_LAUNCHER   "${CCACHE_PROGRAM}")
+  set(CXX_LAUNCHER "${CCACHE_PROGRAM}")
+
+  file(WRITE "${CMAKE_BINARY_DIR}/launch-c" ""
```

Comment:
The launch-c etc. hack is not necessary as we only need/want this working for Makefiles and Ninja, which have official support in CMake for the LAUNCHER method.

Reply:
Thanks a lot for comment @andschwa, I was unsure if there is a plan to add Xcode support at some point.
I updated the PR to remove its usage.
There are two things I'm still not sure of:
- cmake/ccache.cmake : should we introduce it?
- using the file hack makes it easier to export environment variables i.e.: CCACHE_CPP2, CCACHE_DIR or others; should we bother to catch them in CMakeLists at top level or just add them as my current change?